### PR TITLE
[Swift] Add training loop abstraction.

### PR DESCRIPTION
**Pull request by dan-zheng**
Add `train` function, which trains a given model on a given dataset, using
a given optimizer and loss function.

Todos:
- Integrate callbacks into `train`.
  - `train` will be the analog of `class Runner`.
- Finish porting the Python `05b_early_stopping.ipynb` notebook.

### CODE REVIEW

**Comment by dan-zheng** [comment]
Oops, adding reviewers now was a mistake, since this PR is not yet ready for review. Apologies.

File: dev_swift/05b_early_stopping.ipynb [hunk 1]
```diff
@@ -0,0 +1,144 @@
+{
```
Comment (sgugger) [inline dev_swift/05b_early_stopping.ipynb]: <p>It would be great to have a short explanation of the `\Dense&lt;Float&gt;.allDifferentiableVariables` when you have a bit of time.</p>

 _Reply via <a href='https://app.reviewnb.com/fastai/fastai_docs/pull/65/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>
